### PR TITLE
Add support for generic arduino commands

### PR DESCRIPTION
**Pull request by PeterJCLaw**
This adds a new `direct_command` method which takes in a command as a list of JSON-encodable entities and passes that to robotd. It expects to immediately get back a response message (as well as the usual status blob) which contains either valid data from the arduino or a description of the failure.

This includes a novel testing approach which:
- creates a valid robotd Board instance
- mocks the underlying serial connection
- creates a robot-api Board instance
- mocks the socket connecting the robot-api Board such that it
  connects to the robotd Board instance

This avoids real `socket`s and `multiprocessing`, both of which incur substantial overhead during tests along with the introduction of timeouts (which is just nasty).

This relies on https://github.com/sourcebots/robotd/pull/35.

### CODE REVIEW

**Comment by PeterJCLaw** [comment]
The test failure on the CI is due to https://github.com/sourcebots/robotd/pull/35 not being available in that test environment.

**Comment by RealOrangeOne** [comment]
I'm liking this use of fake sockets, hopefully these changes can be backported into the remainder of the tests (separate PR) to reduce the oddness they bring

**Comment by PeterJCLaw** [comment]
@RealOrangeOne indeed, I've some thoughts -> #57

File: robot/servo.py [hunk 1]
```diff
@@ -119,6 +138,34 @@ def serial(self):
...
+        # consume the broadcast status
+        self.receive()
+
+        if status == 'ok':
+            return response['data']
+        else:
+            for cls in (CommandError, InvalidResponse):
+                if cls.__name__ == response['type']:
+                    raise cls(response['description'])
+
```
Comment (kierdavis) [inline robot/servo.py]: What happens if we receive a response with an invalid `type`?

Comment (PeterJCLaw) [reply robot/servo.py]: good point; I'll make that error harder.

Comment (PeterJCLaw) [reply robot/servo.py]: 023496d

**Comment by kierdavis** [comment]
After testing on a kit last night I'm confident this works as intended. I would be grateful if the conflicts could be resolved before I approve.

**Comment by PeterJCLaw** [comment]
@kierdavis conflicts resolved; please could you take another look?

**Pull request merged**